test/integration/server-discovery-and-monitoring/server_discover_and_monitoring.test.ts

```diff
@@ -8,3 +15,112 @@ describe('SDAM Unified Tests (Node Driver)', function () {
   );
   runUnifiedSuite(clonedAndAlteredSpecTests);
 });
+
+describe('Monitoring rtt tests', function () {
+  let client: MongoClient;
+  let heartbeatDurations: Record<string, number[]>;
+  const HEARTBEATS_TO_COLLECT = 200; // Wait for 200 total heartbeats. This is high enough to work for standalone, sharded and our typical 3-node replica set topology tests
+  const IGNORE_SIZE = 20;
+  const DELAY_MS = 10;
+  let count: number;
+  const ee = new EventEmitter();
+
+  const listener = (ev: ServerHeartbeatSucceededEvent) => {
+    if (!client.topology.s.servers.has(ev.connectionId)) return;
+    count++;
+    if (count < IGNORE_SIZE) {
+      return;
+    }
```

Comment:
Just suggestions to pair down the logic here 
- could you just collect all the HBs and then test on a slice of IGNORE_SIZE and up?
- could you inline the listener into the beforeEach and use a promiseWithResolvers to block until you have 200 HBs? then you shouldn't need a seperate ee.emit(done) communication channel

Reply:
I can refactor the tests to collect a smaller number of heartbeats per server and that would enable me to do the slicing.

Reply:
Why do we have to change the total?

Reply:
I had set the number of heartbeats to collect to be 200 total before to be a simple way to get all topologies we test on be able to collect decently large number of heartbeats. That meant that  previously the test was just assuming that each server in the topology would correct roughly the same number of heartbeats, whereas now, we're guaranteeing that we get 65 heartbeats per server, which for the repl set topology works out to 195 heartbeats total, but distributed evenly across each of the servers.

We don't necessarily _need_ to drop the number of heartbeats collected, but collecting 200 heartbeats per server didn't seem necessary.